File: beispiele-einzeln/vererbung/vererbung.java
class Animal {
    public void makeSound() {
        System.out.println("Animal sound");
    }
}

class Dog extends Animal {
    @Override
    public void makeSound() {
        System.out.println("Woof");
    }
}

public class Main {
    public static void main(String[] args) {
        Dog myDog = new Dog();
        myDog.makeSound();
    }
}

// In diesem Beispiel erbt die Klasse Dog 
// von der Klasse Animal und überschreibt die Methode makeSound.

File: beispiele-einzeln/vererbung/vererbung.py
class Animal:
    def make_sound(self):
        print("Animal sound")

class Dog(Animal):
    def make_sound(self):
        print("Woof")

my_dog = Dog()
my_dog.make_sound()

# In diesem Beispiel erbt die Klasse Dog 
# von der Klasse Animal und überschreibt die Methode make_sound.